qiskit_dynamics/signals/signals.py

```diff
@@ -607,12 +572,12 @@ def envelope(t):
             self, envelope=envelope, carrier_freq=carrier_freqs, phase=phases, name=name
         )
 
-    def complex_value(self, t: Union[float, np.array, Array]) -> Union[complex, np.array, Array]:
+    def complex_value(self, t: ArrayLike) -> ArrayLike:
         """Return the sum of the complex values of each component."""
-        if Array.default_backend() == "jax":
-            t = Array(t)
-        exp_phases = np.exp(np.expand_dims(t, -1) * self._carrier_arg + self._phase_arg)
-        return np.sum(self.envelope(t) * exp_phases, axis=-1)
+        exp_phases = unp.exp(
+            unp.expand_dims(unp.asarray(t), -1) * self._carrier_arg + self._phase_arg
```

Comment:
Is `unp.asarray(t)` necessary here, since whatever `t` is, it will just get dispatched by `unp.expand_dims` anyway? Or will this not work if `t` is a list or something.